lib/wasix/src/runtime/task_manager/tokio.rs

```diff
@@ -276,6 +286,8 @@ impl VirtualTaskManager for TokioTaskManager {
                     recycle,
                 });
             });
+
+            rx.recv().unwrap()?;
```

Comment:
There can be weird edge cases where the sender gets dropped, leading to this unwrap() panicking. 
(I guess if new_with_store were to panic)

Can you convert the error returned by `.recv()` and return it instead?

Reply:
[76826d2](https://github.com/wasmerio/wasmer/pull/5440/commits/76826d2b2217cbe0151f130487fc17e1c223a5db)